internal/config/config.go

```diff
@@ -195,21 +197,21 @@ func ParseDSN(dsn string) (UserConfig, error) {
 	}
 	parsedURL, err := url.Parse(fullDSN)
 	if err != nil {
-		return UserConfig{}, errors.Wrap(err, "invalid DSN: invalid format")
+		return UserConfig{}, dbsqlerr.NewRequestError(context.TODO(), dbsqlerr.ErrInvalidDSNFormat, err)
 	}
 	ucfg := UserConfig{}.WithDefaults()
 	ucfg.Protocol = parsedURL.Scheme
 	ucfg.Host = parsedURL.Hostname()
 	port, err := strconv.Atoi(parsedURL.Port())
 	if err != nil {
-		return UserConfig{}, errors.Wrap(err, "invalid DSN: invalid DSN port")
+		return UserConfig{}, dbsqlerr.NewRequestError(context.TODO(), dbsqlerr.ErrInvalidDSNPort, err)
```

Comment:
should we have a context for this function then?

Reply:
ParseDSN is called from databricksDriver.OpenConnector(dsn string) which doesn't take a context.  We can't change OpenConnector() because it is part of the driver.DriverContext interface.